src/server/node_func.c

```diff
@@ -2907,3 +2912,38 @@ chk_vnode_pool (attribute *new, void *pobj, int actmode)
 	}
 	return PBSE_NONE;
 }
+
+/**
+ * @brief
+ * 		action routine for the queue's "partition" attribute
+ *
+ * @param[in]	pattr	-	attribute being set
+ * @param[in]	pobj	-	Object on which attribute is being set
+ * @param[in]	actmode	-	the mode of setting, recovery or just alter
+ *
+ * @return	error code
+ * @retval	PBSE_NONE	-	Success
+ * @retval	!PBSE_NONE	-	Failure
+ *
+ */
+int
+action_node_partition(attribute *pattr, void *pobj, int actmode)
+{
+	struct pbsnode *pnode;
+	pbs_queue 	*pq;
+
+	pnode = (pbsnode *)pobj;
+
+	if (pnode->nd_attr[(int)ND_ATR_Queue].at_flags & ATR_VFLAG_SET) {
+		pq = find_queuebyname(pnode->nd_attr[(int)ND_ATR_Queue].at_val.at_str);
+		if (pq == 0)
+			return PBSE_UNKQUE;
+		if (pq->qu_attr[QA_ATR_partition].at_flags & ATR_VFLAG_SET &&
+				pattr->at_flags & ATR_VFLAG_SET) {
+			if (strcmp(pq->qu_attr[QA_ATR_partition].at_val.at_str, pattr->at_val.at_str) != 0)
+				return PBSE_QUE_NOT_IN_PARTITION;
+		}
+	}
+
+	return PBSE_NONE;
+}
```

Comment:
If I were to guess, I'd suspect the issue with PP-925 is that the action functions are not being called at all.  I didn't suggest fixing this issue now because it would creep the scope of your project.  If you feel now is the right time, I am all for it.  I need it for soft_walltime as well.  I came up with two possibilities.  The first is we pass an additional argument to the action functions which is the entire list of attributes being set.  This means that all the action functions have to do double the work.  They first check if the two attributes are being set together, and if not, they pick one up off the object.  The second idea is to create a copy of the object with all the changes made.  This is problematic because you don't always know what the object is.  There are places where you have function A pass a void* to function B which calls the action function with it.  There is no way for function B to know what type of object it is.  If you have any other ideas, I'd love to hear them.